docs/best_practices.md

```diff
@@ -0,0 +1,153 @@
+
+![Create a Variable](img/create_variable.png)
+
+#### Static Variables
+
+There are two ways to create a static Variable. The first is to provide a simple list of options as comma-separated values (CSV):
+
+```
+value1,value2,value3
+```
+
+The other is to provide a map of values and display names:
+
+```
+display1:value1
+display2:value2
+display3:value3
+```
+
+#### Variables from tags
+
+A common use case is in InfluxDB is to filter data by one of the `tags` in the measurement. You can make this configurable in your Dashboard by making a Variable out of them. Create a Query Variable with a Flux query like this one:
+
+```
+import "influxdata/influxdb/v1"
+v1.measurementTagValues(bucket: "your_bucket_name", measurement: "your_measurement_name", tag: "your_tag_name")
+```
+
+This will give the user a drop-down menu with all of the unique values of the tag `your_tag_name` from the specified measurements.
+
+### Using Labels
+
+[Labels](https://v2.docs.influxdata.com/v2.0/visualize-data/labels/) let you tag your resources for easier identification in the UI, and they will also make it easier to export them into a Template. If you have a mix of resources in your InfluxDB instance you will have to individually specify which ones to export so that you don't get them all.
+
+However, if you create a unique Label to identify those resources, you can use that when exporting so that you don't have to know the IDs of each individual resource you want to include, like this:
+
+```
+influx pkg export all --filter=labelName=your_label_name
+```
+
+## Telegraf Configurations
+
+A good Template needs to be more than just a Dashboard, it should also include ways to send data to a user's InfluxDB instance. This can be done in a number of ways, from making calls directly to the [InfluxDB API](https://v2.docs.influxdata.com/v2.0/write-data/#influxdb-api), using the [InfluxDB Client libraries](https://v2.docs.influxdata.com/v2.0/reference/api/client-libraries/), or by [using Telegraf](https://v2.docs.influxdata.com/v2.0/write-data/use-telegraf/) and it's many plugins that do the bulk of the work for you.
+
+Telegraf is a widely used data collection agent with a large community that has contributed plugins for gathering data from a variety of sources. Whatever data you're collecting, chances are there's a Telegraf plugin that can help you do it. As such we encourage Templates to include a Telegraf configuration for data collection. If you use another method of sending data to InfluxDB, please include that and instructions on using it in your Template's `README.md`.
```

Comment:
Telegraf is a widely used data collection agent with a large community that has contributed plugins for gathering data from a variety of sources. Whatever data you're collecting, chances are there's a Telegraf plugin that can help you do it. ~As such~ We ~encourage~ **recommend** that ~Templates to~ **templates** include a Telegraf configuration for ~data collection~ **collecting data**. If you use another method of sending data to InfluxDB, ~please~ include that **information** and instructions ~on using it~ in your ~Template's~ **template's** `README.md`.